### ISSUE TITLE
Boolean opreation on shapes

### ISSUE THREAD

**Issue by Connor-yang96** [zsviczian/obsidian-excalidraw-plugin#586]
This is not Feature request , bcz it's out of scope for a plugin. and the offical excalidraw[ closed this issue](https://github.com/excalidraw/excalidraw/issues/1906).

But I still have hope for this. 

I don't know how much work for this, I don't know how hard, how difficult for implement this feature. I don't know .

----

**Current solution :**

Make a boolean  shapes on  https://excalidraw-ardpzxmib.vercel.app

![image](https://user-images.githubusercontent.com/81000027/163335223-4ca25c98-702d-44a9-9a33-15f1512f0237.png)

save a file with “xxx.excalidraw" , open file in obsidian. but it's empty.

still can't copy and paste it from website into obsidian. 

image and svg file are not my want.

**Comment by zsviczian** [comment]
This is super cool! I didn't know about this.

My dilemma is the following. I'd love to bring this into Obsidian, because this looks very useful, but I strive to keep Obsidian specific changes to the core Excalidraw package to the bare minimum, because if Excalidraw Obsidian is too different from excalidraw.com a point in time will come when the Obsidian plugin can no longer benefit from the on-going development of the core Excalidraw product, because I will not be able to merge changes into the Obsidian package.

The technical solution in this fork is clever and to a large extent looks very doable with ExcalidrawAutomate / Excalidraw Script engine. What I am thinking to do is to expose required internal Excalidraw functions that will allow someone to build the Excalidraw Automate script in Obsidian to mimic this feature. Overall if I could find the time, I feel this whole thing (exposing functions to the API, and doing the development) would be possible over a weekend... I just don't have the time :(

Loading the file will not work because this introduces a new element type called "path". But, you can export the shape as a transparent SVG and add that to your drawing in Obsidian as an image. That will work nicely.


https://user-images.githubusercontent.com/14358394/163340388-9f93fd3d-d9cd-49b1-acdc-9a8d321126fe.mp4

**Comment by Connor-yang96** [comment]
Okay . I don't push you to do something. I just hope.

If my understanding is not wrong, your means you'll implement it by script ,not in Obsidian package ?

**Comment by dazbog835** [comment]
I never comment on feature requests... I would just like to say that I've been lurking around this boolean feature both here and in the other thread, and I keep getting disappointed by the replies :)

Even just a basic way of cutting away one rectangle from another (which I imagine would be easier than doing it with a circle) would enpower so many other things you could do with excalidraw. I use it inside obsidian and it's incredibly helpful for making maps for my creative writing, quick sketches, etc.

Anyway, such a wonderful plugin. I use it every day for my writing, for my d&d stuff, for designing... If ever there's a plan for adding the boolean featurea, know there are probably more people that would like it :)
Anyway, all the best! I really appreciate all the effort you put into this app and I love your yt channel!

**Comment by zsviczian** [comment]
I still hope to implement this as an Obsidian script... the only issue is, that I have more ideas (much, much, much more) than hours available each day.

**Comment by dazbog835** [comment]
I know, I know, and I appreciate all the time you're putting into this already. Excalidraw is honestly an amazing app. I just wanted to chime in that this would be awesome :) all the best

**Comment by Connor-yang96** [comment]
Is this also belong to "one day maybe " label ? just curious

**Comment by rupadarshiray** [comment]
This is solved by the script [Boolean Operations](https://github.com/zsviczian/obsidian-excalidraw-plugin/blob/master/ea-scripts/Boolean%20Operations.md).

**Comment by zsviczian** [comment]
correct, this was solved a long time ago with the Boolean Operations Script, I just forgot to close this issue. You can find a short demonstration of the script here: https://youtu.be/P_Q6avJGoWI?t=596
